components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadsFeature.kt

```diff
@@ -98,6 +99,14 @@ class DownloadsFeature(
         }
     }
 
+    /**
+     * Notifies the feature that the permissions were denied. It will then
+     * clear the pending download.
+     */
+    fun onPermissionsDenied() {
+        activeSession?.download = Consumable.empty()
+    }
```

Comment:
Is this something we need to do in order to avoid re-requesting the permission when the app resumes (is that something we do?).

I could see an app showing an indicator for a pending download that still needs a permission. But that's something that is planned and we do not need to support it now. And we could add this later by letting the app injecting a different behavior (and ours being the default implementation).

Reply:
Yes, exactly. When the request is denied, I think we should get rid of it to prevent re-requesting it (if someone wants that behaviour) and also to free up the consumable. Mostly found the API missing when I handled permission granted and there was no `else` block.